core/ops.rs

```diff
@@ -68,16 +68,20 @@ impl OpRegistry {
     op_id: OpId,
     control: &[u8],
     zero_copy_buf: Option<PinnedBuf>,
-  ) -> CoreOp {
+  ) -> Result<CoreOp, String> {
     // Op with id 0 has special meaning - it's a special op that is always
     // provided to retrieve op id map. The map consists of name to `OpId`
     // mappings.
     if op_id == 0 {
-      return Op::Sync(self.json_map());
+      return Ok(Op::Sync(self.json_map()));
     }
 
-    let d = &*self.dispatchers.get(op_id as usize).expect("Op not found!");
-    d(control, zero_copy_buf)
+    let d = match self.dispatchers.get(op_id as usize) {
+      Some(handler) => &*handler,
+      None => return Err(format!("Unknown op id: {}", op_id)),
+    };
```

Comment:
It's a little odd for this low-level function to construct an error string like this. I think it might be better if this function returns `Result<CoreOp, ()>` or `Option<CoreOp>` and the caller creates the error message.

Reply:
Also it would be helpful if you add triple slash doc to this function describing when it fails (i.e. it only fails when op id is wrong).